orchestrator/workflows/utils.py

```diff
@@ -40,14 +40,15 @@
 from pydantic_forms.types import FormGenerator, InputForm, InputStepFunc, StateInputStepFunc
 
 
-def _generate_new_subscription_form(workflow_target: str, workflow_name: str) -> InputForm:
+def _generate_new_subscription_form(_workflow_target: str, workflow_name: str) -> InputForm:
     class NewProductPage(FormPage):
         product: ProductId
 
-        @validator("product", allow_reuse=True)
-        def product_validator(cls, v: UUID) -> UUID:  # type: ignore
+        @field_validator("product")
+        @classmethod
+        def product_validator(cls, product_id: UUID) -> UUID:
             """Run validator for initial_input_forms to check if the product exists and that this workflow is valid to run for this product."""
-            product = ProductTable.query.get(v)
+            product = ProductTable.query.get(product_id)
```

Comment:
could use `_validate_is_product` from `orchestrator.forms.validators`

Reply:
That returns the product ID, here we need the product object